views/affectation.py

```diff
@@ -195,12 +195,12 @@ def internship_affectation_sumup(request, cohort_id):
         hospital_specialties[organization.reference] = []
         for offer in offers:
             if offer.organization.reference == organization.reference:
-                informations.append(offer)
+                if(offer.speciality_id == filter_specialty):
```

Comment:
Peut-être remonter cette condition un peu plus haut en utilisant un générateur.
Ex: offers_filtered_by_speciality = filter(lambda offer: offer.speciality_id == filter_specialty, offers)

Et du coup la boucle devient: for offer in _filtered_by_speciality.

Reply:
Ok merci. J'ai utilisé une list comprehension car il fallait alors de toute façon cast en list pour itérer